app.rb

```diff
@@ -0,0 +1,121 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      if person.instance_of?(Student)
+        puts "[Student], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      else
+        puts "[Teacher], ID: #{person.id}, Name: #{person.name}, age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to add a student (1) or a teacher (2)? [Input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    case student_or_teacher
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Please enter a valid number: 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]'
+    permission = gets.chomp.downcase == 'y'
+    student = Student.new(age, permission, name)
+    @people << student
+    puts 'Student created sucessfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    puts 'specialization'
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher
+    puts 'Teacher created sucessfully'
+  end
+
+  def add_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created sucessfully'
+  end
+
+  def add_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Auhtor: #{book.author}"
+    end
+    book_idx = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_idx = gets.chomp.to_i
+
+    print 'Date: '
+    rental_date = gets.chomp
+
+    rental = Rental.new(rental_date, @books[book_idx], @people[person_idx])
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  # def list_all_rentals
+  #   print 'ID of person: '
+  #   id = gets.chomp
+  #   puts 'Rental: '
+  #   @rentals.each do |rental|
+  #     puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
+  #   end
+  # end
+  # Method to list all rentals for a given person ID
```

Comment:
- [ ] **[Optional]** Kindly, you may consider removing these commented lines to keep the codebase clean and improve readability. Uncommented code is easier to understand and maintain. 👍